### PR TITLE
Brew Improvements

### PR DESCRIPTION
**Pull request by nhojb**
## Description

- Improve reliability of `outdated` command
- Add action to copy formula/cask name
- Add cask name & tap to cask details
- Add Toast action to cancel current action
- Add Toast action to copy error log after failure

## Type of change

- Bug fix / improvement for my extension

## Screencast

Screenshot 2022-01-17 at 22 22 41
Screenshot 2022-01-17 at 22 22 46
Screenshot 2022-01-17 at 22 23 25

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

**Comment by nhojb** [comment]
The eslint error appears to be erroneous?

```
/home/runner/work/extensions/extensions/extensions/brew/src/utils.ts
156
   80:7   error  'toast' is never reassigned. Use 'const' instead  prefer-const
```

The variable `toast` is reassigned further down the `showActionsToast()` function. Tslint correctly complains if I change to `const`.

**Review by pernielsentikaer — APPROVED** [review]
Hi @nhojb 

Solid update! I love it, looks good with the new cancel toast 💪 I'm going to accept it and then will my colleague do a final review before it's finally merged.

File: extensions/brew/src/utils.ts [hunk 1]
```diff
@@ -66,9 +66,59 @@ export async function fetchRemote<T>(remote: Remote<T>): Promise<T[]> {
...
+}
+
+export function showActionToast(actionOptions: ActionToastOptions): AbortController | undefined {
+  const options: ToastOptions = {
+    style: ToastStyle.Animated,
+    title: actionOptions.title,
+    message: actionOptions.message
+  };
+
+  let toast: Toast | undefined;
```
Comment (thomaspaulmann) [inline extensions/brew/src/utils.ts:82]: Looks like the linter doesn't like this line and it makes the CI fail. Is there any way to please the linter here? CC @mathieudutour and @grzegorzkrukowski

File: extensions/brew/src/utils.ts [hunk 2]
```diff
@@ -66,9 +66,59 @@ export async function fetchRemote<T>(remote: Remote<T>): Promise<T[]> {
...
+      title: "Cancel",
+      onAction: () => {
+        controller?.abort();
+        toast?.hide();
+      }
+    }
+  }
+
+  toast = new Toast(options);
+  toast.show();
```
Comment (mathieudutour) [inline extensions/brew/src/utils.ts:97]:
```suggestion
  let controller: AbortController | undefined;

  if (actionOptions.cancelable) {
    controller = new AbortController();
    options.primaryAction = {
      title: "Cancel",
      onAction: () => {
        controller?.abort();
        toast?.hide();
      }
    }
  }

  const toast = new Toast(options);
  toast.show();
```

Comment (nhojb) [reply extensions/brew/src/utils.ts:97]: @mathieudutour I don't understand where does the `toast` variable come from on line 90?

But incredibly (and very surprisingly - to me) it works and the `toast` var is valid and non-null.

Comment (mathieudutour) [reply extensions/brew/src/utils.ts]: variable hoisting in javascript 😬

Comment (mathieudutour) [reply extensions/brew/src/utils.ts]: you could also write
```js
const toast = new Toast(options)

if (cancelable) {
  toast.primaryAction = ...
}

toast.show()
```

**Pull request merged**